run/literals.scala has linker *warnings* (pbl in refiner)

I suspect the new scheme for symbol requirements to have a problem somewhere.
```
!! 784 - run/literals.scala                        [output differs]
% diff /localhome/jenkins/a/workspace/scalajs-task-worker/partest/fetchedSources/2.11.7/test/files/run/literals-run.log /localhome/jenkins/a/workspace/scalajs-task-worker/partest/fetchedSources/2.11.7/test/files/run/literals.check
@@ -10,16 +10,3 @@ literals.scala:37: warning: Octal escape literals are deprecated, use \u0000 ins
 literals.scala:37: warning: Octal escape literals are deprecated, use \u0000 instead.
       "\0x61\0x62".getBytes(io.Codec.UTF8.charSet) sameElements Array[Byte](0, 120, 54, 49, 0, 120, 54, 50),
             ^
-Referring to non-existent method java.lang.Integer.compareTo(java.lang.Byte)scala.Int
-  called from java.nio.ByteBuffer.compareTo(java.nio.ByteBuffer)scala.Int
-  called from java.nio.ByteBuffer.equals(java.lang.Object)scala.Boolean
-  called from scala.runtime.BoxesRunTime$.equals(java.lang.Object,java.lang.Object)scala.Boolean
-  called from scala.collection.GenMapLike$class.liftedTree1$1(scala.collection.GenMapLike,scala.collection.GenMap)scala.Boolean
-  called from scala.collection.GenMapLike$class.equals(scala.collection.GenMapLike,java.lang.Object)scala.Boolean
-  called from scala.collection.AbstractMap.equals(java.lang.Object)scala.Boolean
-  called from scala.runtime.BoxesRunTime$.equals(java.lang.Object,java.lang.Object)scala.Boolean
-    (already seen, not repeating call stack)
-involving instantiated classes:
-  java.nio.HeapByteBuffer
-  scala.runtime.BoxesRunTime$
-  scala.collection.mutable.HashMap
```

--- a/tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala
+++ b/tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala
@@ -50,7 +50,9 @@ abstract class GenIncOptimizer private[optimizer] (semantics: Semantics,
     import factory._
 
     callMethods(LongImpl.RuntimeLongClass, LongImpl.AllIntrinsicMethods) ++
-    optional(callMethods(LongImpl.RuntimeLongClass, LongImpl.OptionalIntrinsicMethods))
+    optional(callMethods(LongImpl.RuntimeLongClass, LongImpl.OptionalIntrinsicMethods)) ++
+    callMethods(Definitions.BoxedIntegerClass,
+        Seq("compareTo__jl_Byte__I", "compareTo__jl_Short__I")) // #2184
   }
 
   private[optimizer] val CollOps: AbsCollOps

